l10n_it_intrastat_statement/models/intrastat_statement_purchase_section.py

```diff
@@ -20,7 +20,8 @@ def _prepare_statement_line(self, inv_intra_line, statement_id=None):
         # Amounts
         amount_currency = 0
         if (
-            invoice_id.currency_id != invoice_id.company_id.currency_id
+            invoice_id.partner_id.country_id.currency_id
+            != invoice_id.company_id.currency_id
             and invoice_id.currency_id != self.env.ref("base.EUR")
```

Comment:
Quindi non interessa più `invoice_id.currency_id` ma `invoice_id.partner_id.country_id.currency_id`, come mai?
Pensavo che `amount_currency` andasse valorizzato proprio quando la fattura è in una valuta diversa rispetto alla sua `company`.

Non andrebbe cambiata anche la `invoice_id.currency_id` che viene confrontata con `self.env.ref("base.EUR")`?